src/test/java/tictactoe_game/WinnerTest.java

```diff
@@ -0,0 +1,201 @@
+package tictactoe_game;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static tictactoe_game.model.Element.*;
+
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import tictactoe_game.implementation.WinnerImplementation;
+import tictactoe_game.model.Element;
+import tictactoe_game.model.Grid;
+import tictactoe_game.model.types.*;
+
+public class WinnerTest {
+  @Test
+  @DisplayName("Must win on first row by element `X`")
+  public void must_win_on_first_row_by_x() {
+    Element[][] winGrid = { { X, X, X }, { EMPTY, EMPTY, EMPTY }, { EMPTY, EMPTY, EMPTY } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
+  @DisplayName("Must win on first column by element `X`")
+  public void must_win_on_first_column_by_x() {
+    Element[][] winGrid = { { X, EMPTY, EMPTY }, { X, EMPTY, EMPTY }, { X, EMPTY, EMPTY } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
+  @DisplayName("Must win on diagonal by element `X`")
+  public void must_win_on_diagonal_by_x() {
+    Element[][] winGrid = { { X, EMPTY, EMPTY }, { EMPTY, X, EMPTY }, { EMPTY, EMPTY, X } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
+  @DisplayName("Must win on first row by element `O`")
+  public void must_win_on_first_row_by_o() {
+    Element[][] winGrid = { { O, O, O }, { EMPTY, EMPTY, EMPTY }, { EMPTY, EMPTY, EMPTY } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
+  @DisplayName("Must win on first column by element `O`")
+  public void must_win_on_first_column_by_o() {
+    Element[][] winGrid = { { O, EMPTY, EMPTY }, { O, EMPTY, EMPTY }, { O, EMPTY, EMPTY } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
+  @DisplayName("Must win on diagonal by element `O`")
+  public void must_win_on_diagonal_by_o() {
+    Element[][] winGrid = { { O, EMPTY, EMPTY }, { EMPTY, O, EMPTY }, { EMPTY, EMPTY, O } };
+    Grid newGrid = new Grid(winGrid);
+
+    assertTrue(WinnerImplementation.is_there_a_winner(newGrid));
+  }
+
+  @Test
```

Comment:
The test may pass but not indicate a **draw** since we can replace any element in the grid wit `EMPTY` and still get the same result.

Reply:
See #12